common/src/main/java/com/wynntils/mc/event/ArmSwingEvent.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.mc.event;
+
+import net.minecraft.world.InteractionHand;
+import net.minecraftforge.eventbus.api.Cancelable;
+import net.minecraftforge.eventbus.api.Event;
+
+@Cancelable
+public class ArmSwingEvent extends Event {
+
+    private final ArmSwingContext actionContext;
+
+    private final InteractionHand hand;
+
+    public ArmSwingEvent(ArmSwingContext actionContext, InteractionHand hand) {
+        this.actionContext = actionContext;
+        this.hand = hand;
+    }
+
+    public ArmSwingContext getActionContext() {
+        return actionContext;
+    }
+
+    public InteractionHand getHand() {
+        return hand;
+    }
+
+    public enum ArmSwingContext {
+        DROP_ITEM_FROM_INVENTORY_SCREEN,
+        DROP_ITEM_FROM_HOTBAR,
+        ATTACK_OR_START_BREAKING_BLOCK,
+        BREAKING_BLOCK,
+        INTERACT_WITH_BLOCK,
+        INTERACT_WITH_ENTITY,
+        USE_ITEM,
```

Comment:
This enum is only implemented for one variable, so likely comment out the other values for now.